### PR TITLE
Add machine_helper.go to utils package

### PR DESCRIPTION
**Pull request by csheremeta**
to house functions that may be useful for interacting with machine objects

### CODE REVIEW

**Comment by sam-nguyen7** [comment]
/lgtm
This will make it a lot easier to unit test as well

File: pkg/controller/utils/machine_helper.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package utils
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/prometheus/common/log"
```
Comment (lisa) [inline pkg/controller/utils/machine_helper.go]: How come we're using this for logging and not `sigs.k8s.io/controller-runtime/pkg/log`?

Comment (csheremeta) [reply pkg/controller/utils/machine_helper.go]: This switched on me when I saved the file, and I'm not sure why

Comment (lisa) [reply pkg/controller/utils/machine_helper.go]: Spoooooooky!

File: pkg/controller/utils/machine_helper.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	awsproviderapi "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logf.Log.WithName("machine_helper")
+
+func RemoveLBFromMasterMachines(kclient client.Client, elbName string, masterNodes *machineapi.MachineList) error {
```
Comment (lisa) [inline pkg/controller/utils/machine_helper.go]: Could you add a comment for this function? It should look like:

```
// RemoveLBFromMasterMachines ....
```

Check out [this blog post](https://blog.golang.org/godoc-documenting-go-code) for more on how Go wants code documented.

File: pkg/controller/utils/machine_helper.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		err = updateLBList(kclient, lbList, newLBList, machine, providerSpecDecoded)
+		if err != nil {
+			log.Error(err, "Error updating LB list for machine", "machine", machine.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func AddLBToMasterMachines(kclient client.Client, elbName string, masterNodes *machineapi.MachineList) error {
```
Comment (lisa) [inline pkg/controller/utils/machine_helper.go]: Doc here, too.

File: pkg/controller/utils/machine_helper.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		err = updateLBList(kclient, lbList, newLBList, machine, providerSpecDecoded)
+		if err != nil {
+			log.Error(err, "Error updating LB list for machine", "machine", machine.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func getDecodedProviderSpec(machine machineapi.Machine) (*awsproviderapi.AWSMachineProviderConfig, error) {
```
Comment (lisa) [inline pkg/controller/utils/machine_helper.go]: Less important for non-exported functions/methods, but might still be useful.

Also, this is specific to AWS, so does it make sense to name the function to follow suit?

**Pull request merged**